security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,227 @@
+[asterisc]: https://github.com/ethereum-optimism/asterisc
+[cannon]: https://github.com/ethereum-optimism/optimism/tree/develop/cannon
+[op-program]: https://github.com/ethereum-optimism/optimism/tree/develop/op-program
+
+
+## Failure Modes and Recovery Paths
+
+### Kona and Asterisc Hot Path Failure
+
+- **Description:**
+  The `op-program` + `cannon` dispute game breaks and the `kona` + `asterisc` fault dispute game is placed in the hot path.
+  If `kona` + `asterisc` also fail, the permissioned fallback will need to be activated.
+
+- **Risk Assessment:**
+  Low likelihood, high impact.
+  Low likelihood since `op-program` + `cannon` are already running in production and have meaningful improvements on the way that will increase resilience.
+  If `kona` + `asterisc` fails while in the hotpath, the impact is high since the permissioned fallback will need to be triggered.
+  This engages the security council, requires comms, and causes the chain to fallback to a weaker security model. 
+
+- **Mitigations:** 
+  Offline testing with the VM Runner for both `kona` + `asterisc` as well as `op-program` + `cannon`.
+  The VM Runner also fully tests the integration of the `op-challenger` with these dispute games ([covered in a later section](#-Breaking-Backwards-Compatibility-for-the-op-challenger)).
+  Action tests ensuring both proof systems follow spec.
+
+- **Detection:**
+  The game that broke `op-program` + `cannon` would be invalidated since the respected game type would be switched in the dispute game factory.
+  Once `kona` + `asterisc` is the respected game type, the dispute monitor will watch for all new games and will alert if there is an issue with `kona` + `asterisc`.
+  If `kona` + `asterisc` has an issue while in the hot path, and an invalid output / game is played, the dispute monitor and `op-challenger` would alert and detect invalid games.
+  At this point, the permissioned fallback would need to be engaged since both proof systems are in repair.
+
+- **Recovery Path(s):**
+  Likely the only viable recovery path would be the permissioned fallback unless the `op-program` + `cannon` are fixed and able to be placed back into the hotpath.
+
+
+### Contract Upgrades
+
+- **Description:**
+  The deployment of a new `FaultDisputeGame` instance, `RISCV` VM, and `DelayedWETH` is needed.
+  Two upgrade actions are performed.
+  1. Initialize anchor state for new game type in `AnchorStateRegistry`.
+  2. Set game type of `kona` + `asterisc` to the FDG deployed before the upgrade.
+  The deployment failing is recoverable by simply re-deploying.
+  The upgrade touches components in the hot path so it is a failure mode covered here.
+
+- **Risk Assessment:**
+  Low likelihood, medium impact.
+  The contract upgrades are tested and very simple.
+  Contracts in the hot path are audited and have been in production so are unlikely to have critical bugs.
+  // TODO: impact
+
+- **Mitigations:** 
+
+- **Detection:**
+
+- **Recovery Path(s):**
```

Comment:
@clabby I've added the contract upgrade section but am less familiar with superchain-ops. Do you think you could provide a bit more context here? Basically need to gauge what's the worst failure mode here, considering there could be a bug in the upgrade scripts, but **not** the contracts like the AnchorStateRegistry or the DisputeGameFactory. Is that a good consideration?

Reply:
For this, I think it's generally fine to write it off as a bullet point. There's smart contract upgrade risk in every upgrade we do - don't think we need to get too granular.

Here's a good example: https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-generic-hardfork.md#activation-failure-without-chain-halt-l1-transactions